kotlinx-coroutines-core/jvm/src/scheduling/Tasks.kt

```diff
@@ -9,8 +9,19 @@ import kotlinx.coroutines.internal.*
 import java.util.concurrent.*
 
 
-// Internal debuggability name + thread name prefixes
-internal const val DEFAULT_SCHEDULER_NAME = "DefaultDispatcher"
+/**
+ * Name of the default scheduler.
+ *
+ * Customization provided for when there's no control over used dispatchers/schedulers (i.e when
+ * [Dispatchers.Default] is used within third-party libraries), or when overall default configuration
+ * is acceptable and only the default scheduler name needs to be changed.
+ *
+ * Useful for filtering log messages.
+ */
```

Comment:
Two points to consider. I'm probably overexaggerating with my concerns, I don't think it matters that much, but still I'd like to share my thoughts.

> The names of the worker threads of [Dispatchers.Default] have it as their prefix.

I wanted to write something like this, but it looks like an internal detail that might change (although I doubt it ever will). Moreover, this is indeed used for `Dispatchers.Default`, but it's not limited to that - this name is used as the default `schedulerName` in `CoroutineScheduler`, and there might be other implementations / use cases that will re-use the name (and maybe not only for thread names), making the KDoc slightly outdated

I tried to hint to it with "Useful for filtering log messages", which doesn't give out too many details/specifics, and leaves the door open for other usages.

> This explains why this property is useful for third parties, which is not that useful for internal documentation

That's actually the whole reason why I wrote about the use case and _why_ it can be set at all, I wouldn't do it for public API. 

Without context, you as a maintainer might not understand why this is needed (as demonstrated [by this comment](https://github.com/Kotlin/kotlinx.coroutines/issues/3231#issuecomment-1086127164)), which can lead to unwanted deprecation/removal of this API. I think coroutine maintainers are smart enough to check commit messages and PRs/issues, but still 

__

All in all, let me know what you think, I have no problem with just accepting your current or any other suggestion.

Reply:
Thanks for the insights! Always great to hear when a team has high standards :+1: 

Applied the suggestion